### DISCUSSION TITLE
Version 1.12.0

### DISCUSSION THREAD

**Discussion by anuraaga** [open-telemetry/opentelemetry-java#4235] [Q&A]
This release includes many breaking changes to the metrics SDK as we move towards marking its first stable release.
Notably, if you configure metric `View`s or have written a custom metric exporter, many of the classes and methods will
have been moved or renamed. There are still a few remaining cleanups targeted for the next release after which there
should not be many. Thanks for bearing with us on this.

### API

- New methods have been added to `Context` to propagate context for common Java 8 callback types
- `AttributesBuilder.put` now supports vararg versions for lists with `AttributeKey`
- Multiple metric async callbacks can be registered for the same instrument, and callbacks can be removed

### SDK

- An issue with Android desugaring of the SDK has been worked around
- EXPERIMENTAL: Support for disabling resource keys with `OTEL_EXPERIMENTAL_RESOURCE_DISABLED_KEYS`
- Fixed handling of `schemaUrl` in `Resource.toBuilder()`
- BREAKING: Many changes to `Data` classes used during export
- BREAKING: Many view configuration methods have been removed

#### Metrics

- APIs deprecated in the previous release have been removed
- DEPRECATION: `PrometheusCollector` for exporting OpenTelemetry metrics with the prometheus client library has been deprecated
- EXPERIMENTAL: File-based configuration of views
- Prometheus exporter now supports JPMS modules

#### Logs

- DEPRECATION: `LogData.getName` has been deprecated for removal

<hr /><em>This discussion was created from the release <a href='https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.12.0'>Version 1.12.0</a>.</em>

**Comment by vasigorc** [comment]
Thank you @anuraaga 

We used the same version number property for the following libraries:

```
opentelemetry-api
opentelemetry-exporter-otlp
opentelemetry-extension-annotations
opentelemetry-sdk
```

Of this only `opentelemetry-extension-annotations` is still at `1.11.0`. Has it been moved to another artifact or will it be split henceforth?

**Reply by jkwatson** [reply]
It has been published. It just isn't showing up in all the indexes yet. I just confirmed I could pull it down via a gradle dependency.

**Comment by hdost** [comment]
👋🏼 https://github.com/open-telemetry/opentelemetry-java/pull/4185 As it seems there's a desire for feedback on this.
My first inclination is that mandating a particular HTTP server is restrictive. By limiting consumption of the MetricReader to the Prometheus Collector anyone using alternative HTTP Frameworks can easily keep their applications integrated. 
This is actually my usecase 😓

**Comment by knutwannheden** [comment]
I was not able to find any version of the `opentelemetry-instrumentation-api` and related modules in any `1.12.0-alpha` version. Is there any released version of these modules which is compatible with the 1.12.0 release of `opentelemetry-api`?

**Reply by jkwatson** [reply]
The instrumentation API is released by the instrumentation project and usually lags a week behind the core library release. It should go out this week, I think.
